pkg/shell-api-demo/pages/myCustomPage.vue

```diff
@@ -0,0 +1,106 @@
+
+<script>
+import HelloWorldComponent from '../components/HelloWorld.vue';
+import { NotificationLevel } from '@shell/types/notifications';
+
+export default {
+  name: 'ShellApiDemoPage',
+  mounted () {
+    console.log('Shell API Demo Page mounted', this.$shell);
+  },
+  data() {
+    return {
+      systemApiData: 'No System Info yet. Press the button to fetch it.'
+    }
+  },
+  methods: {
+    slideInTrigger() {
+      this.$shell.slideIn.open(HelloWorldComponent, {
+        title: 'Hello from SlideIn panel!'
+      });
+    },
+    modalTrigger() {
+      this.$shell.modal.open(HelloWorldComponent, {
+        props: { title: 'Hello Modal' }
+      });
+    },
+    notificationTrigger() {
+      this.$shell.notification.send(NotificationLevel.Success, 'Some notification title', 'Hello world! Success!', {});
+    },
+    systemApiTrigger() {
+      this.systemApiData = {
+        gitCommit: this.$shell.system.gitCommit,
+        isDevBuild: this.$shell.system.isDevBuild,
+        isPrereleaseVersion: this.$shell.system.isPrereleaseVersion,
+        isRancherPrime: this.$shell.system.isRancherPrime,  
+        kubernetesVersion: this.$shell.system.kubernetesVersion,
+        rancherVersion: this.$shell.system.rancherVersion,
+        uiVersion: this.$shell.system.uiVersion          
+      }
+    }
+  },
+}
+</script>
+
+<template>
+  <div>
+    <h2>Shell API Demo</h2>
+    <p>This is a custom page to showcase the Shell API capabilities.</p>
+    <br />
+    <br />
+    <button @click="slideInTrigger">Test Slide-in API</button>
```

Comment:
Buttons are missing roles. I'd suggest using RcButton to get all of the styles baked in and to promote using components in extension examples.